### PR TITLE
feat: implement image locality as score plugin

### LINKED ISSUE #86404: Move ImageLocality priority logic to its Score plugin [Migration Phase 2]
/sig scheduling
/priority important-soon

See #86399 as an example.

Part of #85822

### PR DESCRIPTION
**Pull request by draveness**
/sig scheduling
/priority important-soon
/kind feature

Fix #86404 

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/imagelocality/image_locality.go [hunk 1]
```diff
@@ -49,9 +58,15 @@ func (pl *ImageLocality) Score(ctx context.Context, state *framework.CycleState,
 		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("getting node %q from Snapshot: %v", nodeName, err))
 	}
 
-	meta := migration.PriorityMetadata(state)
-	s, err := priorities.ImageLocalityPriorityMap(pod, meta, nodeInfo)
-	return s.Score, migration.ErrorToFrameworkStatus(err)
+	nodeInfos, err := pl.handle.SnapshotSharedLister().NodeInfos().List()
```
Comment (draveness) [inline pkg/scheduler/framework/plugins/imagelocality/image_locality.go]: I think we still need a post-filter plugin for calculating totalNumNodes

cc/ @ahg-g

Comment (ahg-g) [reply pkg/scheduler/framework/plugins/imagelocality/image_locality.go]: You don't need a PreFilter for that, it is just ```len(sharedLister.NodeInfos().List())```

**Comment by ahg-g** [comment]
/approve

I will lgtm once bazel is updated, thanks Draven.

**Comment by ahg-g** [comment]
/lgtm

Thanks!

**Pull request merged**